src/builder.rs

```diff
@@ -67,14 +69,60 @@ pub enum BuildError {
     CircuitError(#[from] circuit_builder::CircuitBuildError),
 }
 
+#[derive(Default)]
+/// Base builder, can generate builders for containers
+pub struct HugrBuilder {
+    base: HugrMut,
+}
+
+impl HugrBuilder {
+    /// Initialize a new builder
+    pub fn new() -> Self {
+        // initially assume to be a module root, will be replaced if not.
+        Self {
+            base: HugrMut::new_module(),
+        }
+    }
+
+    /// Use this builder to build a module HUGR
+    pub fn module_builder(&mut self) -> ModuleBuilder {
+        ModuleBuilder(&mut self.base)
+    }
+
+    /// Use this builder to build a DFG HUGR
+    pub fn root_dfg_builder(
```

Comment:
Should we call it `dfg_builder` (by analogy: `module_builder` returns a `ModuleBuilder`; this returns a `DFGBuilder`)? Edit: hmm, no that's already taken. How about `module_hugr_builder` and `dfg_hugr_builder`?